util/metrics/src/lib.rs

```diff
@@ -1,81 +1,185 @@
+#![allow(missing_docs)]
 //! A lightweight metrics facade used in CKB.
 //!
 //! The `ckb-metrics` crate is a set of tools for metrics.
 //! The crate [`ckb-metrics-service`] is the runtime which handles the metrics data in CKB.
 //!
 //! [`ckb-metrics-service`]: ../ckb_metrics_service/index.html
-//!
-//! ## Use
-//!
-//! The basic use of the facade crate is through the metrics macro: [`metrics!`].
-//!
-//! ### Examples
-//!
-//! ```rust
-//! use ckb_metrics::{metrics, Timer};
-//!
-//! fn run_query(_: &str) -> u64 { 42 }
-//!
-//! pub fn process(query: &str) -> u64 {
-//!     let timer = Timer::start();
-//!     let row_count = run_query(query);
-//!     metrics!(timing, "process.query_time", timer.stop());
-//!     metrics!(counter, "process.query_row_count", row_count);
-//!     row_count
-//! }
-//! ```
-
-use opentelemetry::metrics::Meter;
-use std::time::{Duration, Instant};
-
-#[doc(hidden)]
-pub use opentelemetry as internal;
-
-/// Returns a global meter.
-pub fn global_meter() -> Meter {
-    opentelemetry::global::meter("ckb-metrics")
+
+use lazy_static::lazy_static;
+use prometheus::{
+    register_int_counter, register_int_gauge, register_int_gauge_vec, IntCounter, IntGauge,
+    IntGaugeVec,
+};
+use prometheus_static_metric::make_static_metric;
+
+pub fn metrics_enabled() -> bool {
+    match ENABLE_METRICS_SERVICE.get() {
+        Some(&enabled) => enabled,
+        None => false,
+    }
+}
+
+/// Indicate whether the metrics service is enabled.
+pub static ENABLE_METRICS_SERVICE: once_cell::sync::OnceCell<bool> =
```

Comment:
I'm not sure whether thread local is faster.

The thread local variable is a `Option<bool>`. When it is None, get the value from `OnceCell` and save the value. If it is some, we don't need to access OnceCell.

Reply:
If we declare a thread_local! variable, it should be mutable, then we need Mutex to protect the thread_local! variable.

Reply:
/// Indicate whether the metrics service is enabled.
pub static ENABLE_METRICS_SERVICE: once_cell::sync::OnceCell<bool> =
    once_cell::sync::OnceCell::new();

thread_local! {
    static ENABLE_COLLECT_METRICS: Option<bool> = None;
}

pub fn metrics_enabled() -> bool {
    ENABLE_COLLECT_METRICS.with(|enabled| enabled.unwrap_or_else(||{
        // how to update the thread_local value?
        false
    }))
}

Reply:
Or use RefCell?
```rust
use std::cell::RefCell;

/// Indicate whether the metrics service is enabled.
pub static ENABLE_METRICS_SERVICE: once_cell::sync::OnceCell<bool> =
    once_cell::sync::OnceCell::new();

thread_local! {
    static ENABLE_COLLECT_METRICS: RefCell<Option<bool>> = RefCell::default();
}

pub fn metrics_enabled() -> bool {
    ENABLE_COLLECT_METRICS.with(|cell| {
        cell.borrow()
            .unwrap_or_else(|| match ENABLE_METRICS_SERVICE.get() {
                Some(enabled) => {
                    cell.replace(Some(*enabled));
                    *enabled
                }
                None => false,
            })
    })
}
```

Reply:
https://docs.rs/thread_local/latest/thread_local/ 
try this one.

Reply:
Since `Option<bool>` is `Copy`, `Cell<Option<bool>>` should work.